### PR TITLE
Switch from vendoring PCRE2 to downloading and building it

### PR DESCRIPTION
**Pull request by ridiculousfish**
As discussed in #8355 and also in #2985, this removes the vendored PCRE2 sources. The `FISH_USE_SYSTEM_PCRE2` CMake variable (if false) now downloads and build PCRE2 directly, using the [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) module.

This should probably go in after a 3.4 release.

### FetchContent

CMake's FetchContent module works by checking out a git repo at configure time, by default into `_deps/pcre2-src` under the build directory. This directory is 18.5 MB. We then use `add_subdirectory` as normal to pull in PCRE2's CMake build.

### Advantages

- Updating to a later PCRE2 just means changing the commit hash
- Sources are a lot smaller
- Plays nicely with the Mac build, since we can build PCRE2 as universal (ARM + x86)

### Disadvantages

- This mode now requires GitHub to be accessible on the Internet for the initial configure
- This mode requires CMake 3.11 (released March 2018)

I don't know how to evaluate the impact on Linux distros that use the vendored PCRE2 like Arch, but I think we want them to stop doing that anyways.

### CODE REVIEW

File: build_tools/make_pkg.sh [hunk 1]
```diff
@@ -27,7 +27,7 @@ SRC_DIR=$PWD
 OUTPUT_PATH=${FISH_ARTEFACT_PATH:-~/fish_built}
 
 mkdir -p "$PKGDIR/build" "$PKGDIR/root" "$PKGDIR/intermediates" "$PKGDIR/dst"
-{ cd "$PKGDIR/build" && cmake -DMAC_INJECT_GET_TASK_ALLOW=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo -DWITH_GETTEXT=OFF -DCMAKE_OSX_ARCHITECTURES='arm64;x86_64' -DMAC_CODESIGN_ID="${MAC_CODESIGN_ID}" "$SRC_DIR" && make VERBOSE=1 -j 12 && env DESTDIR="$PKGDIR/root/" make install; }
+{ cd "$PKGDIR/build" && cmake -DMAC_INJECT_GET_TASK_ALLOW=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo -DWITH_GETTEXT=OFF -DFISH_USE_SYSTEM_PCRE2=OFF -DCMAKE_OSX_ARCHITECTURES='arm64;x86_64' -DMAC_CODESIGN_ID="${MAC_CODESIGN_ID}" "$SRC_DIR" && make VERBOSE=1 -j 12 && env DESTDIR="$PKGDIR/root/" make install; }
```
[the remark this answers is not in the document]
Comment (ridiculousfish) [reply build_tools/make_pkg.sh:30]: done

File: cmake/PCRE2.cmake [hunk 2]
```diff
@@ -28,16 +28,34 @@ else()
...
+  include(FetchContent)
+  set(CMAKE_TLS_VERIFY true)
+  set(PCRE2_REPO "https://github.com/PhilipHazel/pcre2.git")
+
+  message(STATUS "Fetching and configuring PCRE2 from ${PCRE2_REPO}")
+  Set(FETCHCONTENT_QUIET FALSE)
+  FetchContent_Declare(
+    pcre2
+    GIT_REPOSITORY ${PCRE2_REPO}
+    GIT_TAG "72669190cb947f0cac1d038a8bb1820da59ef447" # tag: pcre2-10.36
```
[the remark this answers is not in the document]
Comment (ridiculousfish) [reply cmake/PCRE2.cmake:47]: Staying with the specific git hash per CMake guidance.

**Comment by ridiculousfish** [comment]
Bravely merging.

**Comment by floam** [comment]
We can make the clone a little lighter-weight and slow down the CMake generation step less with GIT_SHALLOW 1.

**Comment by zanchey** [comment]
CentOS/RHEL 7 does have PCRE2 packages - they were added sometime after release. I'll add them to the spec file and it should be ok (there's another issue causing breakage on old PCRE2 at present).

**Comment by floam** [comment]
GIT_SHALLOW does make lighten the load but for some reason not quite as significantly as I suspected it ought to. Looks like [a bug being tracked by CMake](https://gitlab.kitware.com/cmake/cmake/-/issues/17770).

**Pull request merged**